node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/SslConnectionSocketFactoryUpdater.java

```diff
@@ -1,91 +1,86 @@
 
+import com.yahoo.vespa.athenz.api.AthenzIdentity;
+import com.yahoo.vespa.athenz.api.AthenzService;
+import com.yahoo.vespa.athenz.identity.SiaIdentityProvider;
+import com.yahoo.vespa.athenz.tls.AthenzIdentityVerifier;
+import com.yahoo.vespa.athenz.tls.SslContextBuilder;
+import com.yahoo.vespa.athenz.utils.AthenzIdentities;
 import com.yahoo.vespa.hosted.node.admin.component.ConfigServerInfo;
-import com.yahoo.vespa.hosted.node.admin.configserver.certificate.ConfigServerKeyStoreRefresher;
-import com.yahoo.vespa.hosted.node.admin.configserver.certificate.ConfigServerKeyStoreRefresherFactory;
 import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
 
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLContext;
+import java.io.File;
+import java.nio.file.Paths;
 import java.util.HashSet;
-import java.util.Optional;
 import java.util.Set;
 
+import static java.util.Collections.singleton;
+
 /**
- * Responsible for updating SSLConnectionSocketFactory on ConfigServerApiImpl asynchronously
- * and as required by embedded certificate expiry
+ * Responsible for updating {@link SSLConnectionSocketFactory} on {@link ConfigServerApiImpl} asynchronously
+ * using SIA based certificates (through {@link SiaIdentityProvider}).
  *
+ * @author bjorncs
  * @author hakon
  */
 public class SslConnectionSocketFactoryUpdater implements AutoCloseable {
-    private final ConfigServerInfo configServerInfo;
-    private final SslConnectionSocketFactoryCreator socketFactoryCreator;
-    // Internal ConfigServerApi used to refresh the key store
-    private final ConfigServerApiImpl configServerApi;
-    private final Optional<ConfigServerKeyStoreRefresher> keyStoreRefresher;
 
     private final Object monitor = new Object();
-    private SSLConnectionSocketFactory socketFactory = null;
+    private final HostnameVerifier configServerHostnameVerifier;
+    private final SiaIdentityProvider sia;
+
     private final Set<ConfigServerApi> configServerApis = new HashSet<>();
+    private SSLConnectionSocketFactory socketFactory;
 
     /**
      * Creates an updater with valid initial {@link SSLConnectionSocketFactory}
      *
-     * @param hostname         the hostname of localhost
      * @throws RuntimeException if e.g. key store options have been specified, but was unable
      *                          create a create a key store with a valid certificate
      */
-    public static SslConnectionSocketFactoryUpdater createAndRefreshKeyStoreIfNeeded(
-            ConfigServerInfo configServerInfo, String hostname) {
-        return new SslConnectionSocketFactoryUpdater(
-                configServerInfo,
-                hostname,
-                ConfigServerKeyStoreRefresher::new,
-                new SslConnectionSocketFactoryCreator());
+    public static SslConnectionSocketFactoryUpdater createAndRefreshKeyStoreIfNeeded(ConfigServerInfo configServerInfo) {
+        SiaIdentityProvider siaIdentityProvider = configServerInfo.getSiaConfig()
```

Comment:
If you're unconditionally replacing the former implementation, you also need to fill the prerequisite sia config in ConfigServerConfig everywhere.

Reply:
See PR on the _hosted_ repository. Afaik, all places where that config is generated should populate the new parameters.

Reply:
Sorry, for some reason I missed that PR.